grype/matcher/apk/matcher.go

```diff
@@ -23,39 +26,109 @@ func (m *Matcher) Type() match.MatcherType {
 func (m *Matcher) Match(store vulnerability.Provider, d *distro.Distro, p pkg.Package) ([]match.Match, error) {
 	var matches = make([]match.Match, 0)
 
-	// map {  CVE string : []match }
-	var secDbCandidates = make(map[string][]match.Match)
-
 	// find Alpine SecDB matches for the given package name and version
 	secDbMatches, err := common.FindMatchesByPackageDistro(store, d, p, m.Type())
 	if err != nil {
 		return nil, err
 	}
 
-	for _, secDbMatch := range secDbMatches {
-		secDbCandidates[secDbMatch.Vulnerability.ID] = append(secDbCandidates[secDbMatch.Vulnerability.ID], secDbMatch)
+	cpeMatches, err := m.cpeMatchesWithoutSecDbFixes(store, d, p)
+	if err != nil {
+		return nil, err
+	}
+
+	// keep all secdb matches, as this is an authoritative source
+	matches = append(matches, secDbMatches...)
+
+	// add additional unique matches from CPE source that is unique from the SecDB matches
+	secDbMatchesByID := matchesByID(secDbMatches)
+	cpeMatchesByID := matchesByID(cpeMatches)
+	for id, cpeMatchesForID := range cpeMatchesByID {
```

Comment:
This `for` block could be broken out as its own function as an opportunity to express a noteworthy piece of the set operation — i.e. "get the CPE matches not found in Alpine SecDB". That function could return a slice that gets appended to the caller's match slice.